components/Image/image-responsive.tsx

```diff
@@ -0,0 +1,288 @@
+      let offLeft: number
+      let offTop: number
+      let height: number
+      let width: number
+      img.current ? (height = img.current.naturalHeight) : (height = 0)
+      img.current ? (width = img.current.naturalWidth) : (width = 0)
+      img.current ? (offLeft = img.current?.offsetLeft) : (offLeft = 0)
+      img.current ? (offTop = img.current?.offsetTop) : (offTop = 0)
+      //Top and left Limits are negative numbers: we are translationg up and left and screen plan is positive down and right on computers
+      setTopLimit(-(height - fHeight))
+      setLeftLimit(-(width - fWidth))
+      setOrgHeight(height)
+      setOrgWidth(width)
+      setScHeight(height)
+      setScWidth(width)
+      setOffsetLeft(offLeft)
+      setOffsetLeft(offTop)
+    }
+  }
+
+  //After every change in scale, we need to change scaling width,height and translate to new position near pinch
+  const setNewImageLimits = (scale: number, x: number, y: number) => {
+    const img = imgRef
+    let newTopLimit
+    let newLeftLimit
+    let newSCHeight = Math.round(orgHeight * scale)
+    let newSCWidth = Math.round(orgWidth * scale)
+    newTopLimit = -(newSCHeight - fHeight)
+    newLeftLimit = -(newSCWidth - fWidth)
+    setScHeight(newSCHeight)
+    setScWidth(newSCWidth)
+    setTopLimit(newTopLimit)
+    setLeftLimit(newLeftLimit)
+    let newLeft = Math.round((scale / imgScale) * imgLeft) //not exactly correct scaled pinch X origin but best we can do now
+    let newTop = Math.round((scale / imgScale) * imgTop) //not exactly correct scaled pinch Y origin but best we can do now
+    //New left coord cannot be less than new left limit or greater than 0
+    if (newLeft < newLeftLimit) {
+      setImgLeft(newLeftLimit)
+    } else if (newLeft >= 0) {
+      setImgLeft(0)
+    } else setImgLeft(newLeft)
+    //New top coord cannot be less than new top limit or greater than 0
+    if (newTop < newTopLimit) {
+      setImgTop(newTopLimit)
+    } else if (newTop >= 0) {
+      setImgTop(0)
+    } else setImgTop(newTop)
+  }
+
+  const setImageSource = () => {
+    switch (sourceType) {
+      //the source is a base64 string itself...
+      case 1:
+        setSource(`data:image/jpeg;base64,${imgSource}`)
+        break
+      //...or it is the key to an image stored in our images context object...
+      case 2:
+        setSource('data:image/jpeg;base64,' + images[imgSource])
+        break
+      //...or it's a plain url
+      default:
+        setSource(imgSource)
+        break
+    }
+  }
+
+  useEffect(() => {
+    setImageSource()
+    if (imgRef.current?.complete) {
+      //always need to check if image has actually loaded (given NextJS behavior)
+      handleImageLoad()
+      setIsLoaded(true)
+    }
+  }, [])
+
+  useEffect(() => {}, [scWidth])
+
+  /* POINT AND DISTANCE CALCULATION FUNCTIONS*/
+
+  //function called when we detect more than one touch. Assuming pinch only between first two touched detected
+  function distance(e: any) {
```

Comment:
Let's get a type setup for this event. See: https://fettblog.eu/typescript-react/events/